### PR TITLE
🔧 burn collection all chains

### LINKED ISSUE #7875: Burn or Destroy collection not available on kodadot Artists are unable to withdraw Reserved balance
### Is your feature request related to a problem?

While Creating a collection on Asset Hub we inform users that their Deposit is refundable but they are unable to get the refund so there should be a destroy button on collection page

### Describe the solution you would like

Destroy button

### Describe alternatives you have considered

nothing

### Screenshots

not avaliable

### Are there opened related issues?

https://github.com/kodadot/nft-gallery/issues/7097

### PR DESCRIPTION
**Pull request by roiLeo**
- close #7875 

note @preschian made some changes regarding #7908

### CODE REVIEW

**Comment by roiLeo** [comment]
> image

rmrk2 or v1? we only have burn collection tx for assethub right now

**Comment by prury** [comment]
> > image
> 
> rmrk2 or v1? we only have burn collection tx for assethub right now

1, 2 and BSX, so, are you just adding back the delete button back, right?

**Comment by roiLeo** [comment]
> 1, 2 and BSX, so, are you just adding back the delete button, right?

delete and report, I'll be adding burn tx on each chain gradually

**Comment by prury** [comment]
> > 1, 2 and BSX, so, are you just adding back the delete button, right?
> 
> delete and report, I'll be adding burn tx on each chain gradually

nice, then i would just make the delete button disabled just in case it reaches prod
image

**Comment by prury** [comment]
RMRK1 - 
transaction goes but koda does not reflect burning
https://kusama.subscan.io/tx/0x7acfc36f0ef76223321a182bccb5b139bcaa847f6eaad230b64ae3cf87ee37a8
https://deploy-preview-7960--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-TECH_CAVEM

BSX -> wallet never asks for signing 

Snek -> 
image

**Comment by roiLeo** [comment]
> RMRK1 - transaction goes but koda does not reflect burning 

you have 1 item in this collection so it won't work, you need to empty the collection before destroy
Just saw that we miss `burn` on Collection entity

> BSX -> wallet never asks for signing
> 
> Snek ->

This two chain should be the same, please check that all nft from the collection are burned and that you have no ongoing offers.

File: composables/transaction/transactionBurn.ts [hunk 1]
```diff
@@ -108,12 +108,45 @@ export function execBurnCollection(
...
+        //     payload: { id: params.collectionId.toString() },
+        //   }),
+        // ],
+      })
+    }
+
+    if (params.urlPrefix === 'snek' || params.urlPrefix === 'bsx') {
+      executeTransaction({
+        cb: api.tx.nft.destroyCollection,
+        arg: [params.collectionId.toString()],
```
Comment (daiagi) [inline composables/transaction/transactionBurn.ts:140]: Nicer if conversion is too string will happens in only 1 place

**Review by daiagi — APPROVED** [review]
Code lgtm

**Comment by prury** [comment]
> > RMRK1 - transaction goes but koda does not reflect burning
> 
> you have 1 item in this collection so it won't work, you need to empty the collection before destroy Just saw that we miss `burn` on Collection entity
> 
> > BSX -> wallet never asks for signing
> > Snek ->
> 
> This two chain should be the same, please check that all nft from the collection are burned and that you have no ongoing offers.

RMRK ->
removed the item https://kusama.subscan.io/tx/0x7dc86e27bb972885da3dea0434783e2aab5ac3e8d5957d144387b42853908b2b 
tried to burn the collection, but nothing happened to it : https://kusama.subscan.io/tx/0x01d5287282d2360f23e881d7fba00c46a02cbc939c8e6849e30bc56c470c2ab1

BSX -> https://basilisk.subscan.io/tx/0x760f2fb0d861e7681b11f86e11983e2c880b227d55c09bf3fa7ad77c83bb0be6
Success

File: composables/transaction/transactionBurn.ts (hunk 1 above)
Comment (roiLeo) [reply composables/transaction/transactionBurn.ts:140]: oh you mean to type string at the top of the function?

Comment (daiagi) [reply composables/transaction/transactionBurn.ts:140]: The repeating `collectionId.toString() `

**Comment by roiLeo** [comment]
> RMRK -> removed the item https://kusama.subscan.io/tx/0x7dc86e27bb972885da3dea0434783e2aab5ac3e8d5957d144387b42853908b2b tried to burn the collection, but nothing happened to it : https://kusama.subscan.io/tx/0x01d5287282d2360f23e881d7fba00c46a02cbc939c8e6849e30bc56c470c2ab1

tx looks oki, we have no way of knowing if the collection is burnt inside app. (even on collection select) maybe try to mint something inside, if you can mint then the collection isn't burned

> BSX -> https://basilisk.subscan.io/tx/0x760f2fb0d861e7681b11f86e11983e2c880b227d55c09bf3fa7ad77c83bb0be6 Success
> 
> SNEK is acting different now, it never asks for signing

Noice! snek should have same behavior as bsx

**Comment by prury** [comment]
> tx looks oki, we have no way of knowing if the collection is burnt inside app. (even on collection select) maybe try to mint something inside, if you can mint then the collection isn't burned

collection creation(fresh one) - [Create Collection](https://kusama.subscan.io/tx/0x0fe6a1b06aa3dfd42150775a0b619b863588b62e724a06cf540917d975500f42)
collection burn(waited for some blocks pass before burning): [Burn](https://kusama.subscan.io/tx/0xf7ddf4018d2d91b79868a2ca92622d297da78703e4150182665d330e92e942d4)
collection mint: 
[Mint NFT inside burned collection](https://kusama.subscan.io/tx/0xa2f62be61737683cba4368686637572c5ffe0aaffc03ae63b20ae1bcd7a4bed7) [Collection link](https://deploy-preview-7960--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-WGUEGSYF2DOWV)


> 

> > SNEK is acting different now, it never asks for signing
> 
> Noice! snek should have same behavior as bsx

signing being asked now

File: composables/transaction/transactionBurn.ts [hunk 2]
```diff
@@ -107,13 +107,42 @@ export function execBurnCollection(
...
+    }
 
-    executeTransaction({
-      cb,
-      arg: [params.collectionId.toString(), {}],
-    })
+    if (params.urlPrefix === 'ksm') {
+      executeTransaction({
+        cb: api.tx.system.remark,
+        arg: [collectionId, {}],
```
Comment (preschian) [inline composables/transaction/transactionBurn.ts:123]: is this how to destroy collection on ksm?

**Review by preschian — APPROVED** [review]

**Comment by preschian** [comment]
oh, one thing, I forgot to add `$updateLoader(true)` in here https://github.com/roiLeo/nft-gallery/blob/ae1befadd2c94c86b632e31cbe5904e7ba034ba2/components/collection/HeroButtons.vue#L155

File: composables/transaction/transactionBurn.ts (hunk 2 above)
Comment (roiLeo) [reply composables/transaction/transactionBurn.ts:123]: yes, but idk if it's working I think I need kodadot/packages#180 first

ref https://github.com/rmrk-team/rmrk-substrate/blob/8c74166627c496435e2dff5dbea03a55d8fec764/pallets/rmrk-core/src/lib.rs#L574

**Comment by roiLeo** [comment]
> oh, one thing, I forgot to add `$updateLoader(true)` in here 

you mean `(false)`?

**Pull request merged**